src/wit_gen.rs

```diff
@@ -71,11 +71,18 @@ impl<F> TraceContext<F> {
 
 pub type Trace<F, TraceArgs> = dyn Fn(&mut TraceContext<F>, TraceArgs) + 'static;
 
-#[derive(Clone)]
 pub struct TraceGenerator<F, TraceArgs> {
     trace: Rc<Trace<F, TraceArgs>>,
 }
 
+impl<F, TraceArgs> Clone for TraceGenerator<F, TraceArgs> {
+    fn clone(&self) -> Self {
+        Self {
+            trace: self.trace.clone(),
+        }
+    }
+}
+
```

Comment:
Same question.

Reply:
The derived implementation requires TraceArgs to be Clone, when it is not necessary.

Reply:
Got it. A bit surprised that you could work around the requirement like this. Learned something new today. ;)